iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/MoveFeederBay.java

```diff
@@ -0,0 +1,132 @@
+    }
+
+    @Override
+    public void apply(Network network, NamingStrategy namingStrategy, boolean throwException, ComputationManager computationManager, ReportNode reportNode) {
+        Connectable<?> connectable = network.getConnectable(connectableId);
+        if (!checkConnectable(throwException, reportNode, connectable)) {
+            return;
+        }
+        VoltageLevel voltageLevel = network.getVoltageLevel(targetVoltageLevelId);
+        if (voltageLevel.getTopologyKind() == TopologyKind.NODE_BREAKER) {
+            cleanNodeBreakerTopology(network, connectableId, reportNode);
+            createTopology(getSideFromTerminal(terminal, connectable), targetBusOrBusbarSectionId, network, voltageLevel, connectable, namingStrategy, reportNode);
+            Map<VoltageLevel, Integer> firstAvailableNodes = new HashMap<>();
+            int connectableNode = firstAvailableNodes.compute(voltageLevel, this::getNextAvailableNode);
+            terminal.getNodeBreakerView().moveConnectable(connectableNode, voltageLevel.getId());
+        } else {
+            terminal.getBusBreakerView().moveConnectable(targetBusOrBusbarSectionId, terminal.isConnected());
+        }
+    }
+
+    private int getSideFromTerminal(Terminal terminal, Connectable<?> connectable) {
+        if (connectable instanceof Injection<?>) {
+            return 0;
+        } else if (connectable instanceof Branch<?>) {
+            return terminal.getSide().getNum();
+        }
+        throw new IllegalStateException("Unexpected connectable: " + connectable);
+    }
+
+    private int getNextAvailableNode(VoltageLevel vl, Integer node) {
+        return node == null ? vl.getNodeBreakerView().getMaximumNodeIndex() + 1 : node + 1;
+    }
+
+    @Override
+    public String getName() {
+        return "MoveFeederBay";
+    }
+
+    @Override
+    public NetworkModificationImpact hasImpactOnNetwork(Network network) {
+        NetworkModificationImpact impact = DEFAULT_IMPACT;
+
+        // Check connectable and target busbar section exists
+        Connectable<?> connectable = network.getConnectable(connectableId);
+        BusbarSection targetBusbarSection = network.getBusbarSection(targetBusOrBusbarSectionId);
+        if (connectable == null || connectable instanceof BusbarSection || targetBusbarSection == null) {
+            impact = NetworkModificationImpact.CANNOT_BE_APPLIED;
+            return impact;
+        }
+
+        // Check that at least one terminal is in the same voltage level as the target busbar section
+        VoltageLevel targetVoltageLevel = targetBusbarSection.getTerminal().getVoltageLevel();
+        boolean hasTerminalInTargetVoltageLevel = false;
+
+        for (Terminal terminal : connectable.getTerminals()) {
+            if (terminal.getVoltageLevel().getId().equals(targetVoltageLevel.getId())) {
+                hasTerminalInTargetVoltageLevel = true;
+                break;
+            }
+        }
```

Comment:
This can be replaced by:
```suggestion
        boolean hasTerminalInTargetVoltageLevel = connectable.getTerminals().stream()
                .map(t -> t.getVoltageLevel().getId())
                .anyMatch(id -> id.equals(targetVoltageLevel.getId()));
```